src/main/java/org/truffleruby/parser/ast/SymbolParseNode.java

```diff
@@ -55,14 +55,14 @@
 
     // Interned ident path (e.g. [':', ident]).
     public SymbolParseNode(SourceIndexLength position, String name, Encoding encoding, CodeRange cr) {
-        // The cr argument is ignored since it is not always correct
         super(position);
         this.name = name;  // Assumed all names are already intern'd by lexer.
-        if (encoding.isAsciiCompatible() && StringOperations.isASCIIOnly(name)) {
-            this.rope = StringOperations.encodeRope(name, USASCIIEncoding.INSTANCE, CodeRange.CR_7BIT);
-        } else {
-            this.rope = StringOperations.encodeRope(name, encoding, CodeRange.CR_UNKNOWN);
+
+        if (encoding == USASCIIEncoding.INSTANCE || cr == CodeRange.CR_7BIT) {
+            encoding = USASCIIEncoding.INSTANCE;
```

Comment:
The first `encoding == USASCIIEncoding.INSTANCE` condition seems extra here.

Reply:
Should it be `encoding.isAsciiCompatible()`? Or does cr == 7-bit implies that? (it would be good to clarify with a comment in such a case).

Reply:
You're both reviewing a revert here. It's fine, but we've never done that before. I'd really rather commit the revert to get back to our previous state and then make any improvements.

We can investigate whether the encoding check is really necessary. I believe it's there in the even the code range couldn't be determined (i.e., `CR_UNKNOWN`), but that may no longer be possible.

`CR_7BIT` is only applied to ASCII-compatible encodings. That's why I updated the lexer to make that clearer. I can add a comment, but this implication is made throughout our entire codebase.

Reply:
@bjfish fix improved the code so I'd rather keep it clean :)

> I can add a comment, but this implication is made throughout our entire codebase.

Could add a comment in the CodeRange class that CR_7BIT is only applied for US-ASCII-compatible encodings (so it's not just all bytes are < 128).
Then we can simplify this check to just `cr == CodeRange.CR_7BIT`.
Except that is wrong if `cr` is null :( Can it be null?

Reply:
The first condition in the if does this:
```
if(encoding == USASCIIEncoding.INSTANCE) {
  encoding = USASCIIEncoding.INSTANCE;
}
```
So, isn't that condition just no-op?  Maybe the intention here was to do something like this?
```
if(encoding != USASCIIEncoding.INSTANCE && cr == CodeRange.CR_7BIT) {
  encoding = USASCIIEncoding.INSTANCE;
}
```

Reply:
Unsure. It could have been there to force the `cr` to `CR_7BIT` and that part dropped off for whatever reason. I'm tidying it up a bit differently right now.